src/main/java/eu/dissco/backend/service/DigitalMediaObjectService.java

```diff
@@ -30,14 +30,23 @@ public JsonApiWrapper getDigitalMediaByIdJsonResponse(String id, String path) {
     return new JsonApiWrapper(dataNode, linksNode);
   }
 
-  public JsonApiMetaWrapper getAnnotationsOnDigitalMediaObject(String id, String path,
+  public JsonApiListResponseWrapper getAnnotationsOnDigitalMediaObject(String id, String path,
       int pageNumber, int pageSize) {
     String mediaId = "https://hdl.handle.net/" + id;
-    var annotations = repository.getAnnotationsOnDigitalMediaObject(mediaId);
-    int totalPageCount = repository.getAnnotationPageCountOnMediaObject(mediaId, pageSize);
-    var metaNode = new JsonApiMeta(totalPageCount);
-    var linksNode = buildLinksNode(path, pageNumber, pageSize, totalPageCount);
-    return new JsonApiMetaWrapper(annotations, linksNode, metaNode);
+    var annotationsPlusOne = repository.getAnnotationsOnDigitalMediaObject(mediaId, pageNumber, pageSize+1);
+
+    boolean hasNextPage;
```

Comment:
I would probably put this either in a parent or somewhere else as the same lines seem to be used in multiple classes